lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,374 @@
+"use strict";
+
+const EMIT_DATA = "data", EMIT_ERR = "error";
+
+module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
+    var cycleCounter, writeRecord;
+
+    /**
+     * Updates heartbeatDate of a specific record and calls the callback that is
+     * passed in.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function heartbeat(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`UPDATE ??
+            SET heartbeatDate = NOW()
+            WHERE hash = ?
+                AND isProcessing = true
+                AND owner = ?;`,
```

Comment:
Did you finish that SQL guide?

Does the semicolon need to exist?

I know I said indentation before, but I've been reconsidering it because of how subselects would get embedded.  Here's something with some imaginary scheme

    SELECT user, email
    FROM tableOne
        LEFT JOIN (
            SELECT id, email
            FROM emailTable
        ) AS emailSelect
    ON tableOne.emailId = emailSelect.id
    WHERE user LIKE "%a%"
        AND email LIKE "%a%"
        AND user IN (
            SELECT DISTICT user
            FROM badUserTable
        )
    ORDER BY user DESC

I could easily imagine that the rule of "indent only inside subselects" or similar would be useful.